server/middleware/verifyAutomationScheduler.js

```diff
@@ -1,10 +1,25 @@
-const verifyAutomationScheduler = (req, res, next) => {
+const {
+  getCollectionJobById
+} = require('../models/services/CollectionJobService');
+
+const verifyAutomationScheduler = async (req, res, next) => {
   const incomingSecret = req.headers['x-automation-secret'];
+  const { jobID: id } = req.params;
+
+  if (!id) return res.status(404).json({ error: 'unknown jobId param' });
+  const job = await getCollectionJobById({
+    id,
+    transaction: req.transaction
+  });
+  if (!job)
+    return res
+      .status(404)
+      .json({ error: `Could not find job with jobId: ${id}` });
+
+  // store the collection job on the request to avoid a second lookup
+  req.collectionJob = job;
```

Comment:
I was wondering how the `collectionJob` was going to get on the `req` when I looked at `AutomationController` before this. Thanks for the comment explainer